prow/cluster/build/prowjob_service_accounts.yaml

```diff
@@ -11,6 +11,16 @@ metadata:
 ---
 apiVersion: v1
 kind: ServiceAccount
+metadata:
+  annotations:
+    iam.gke.io/gcp-service-account: istio-prow-test-job-private@istio-testing.iam.gserviceaccount.com
+  labels:
+    app.kubernetes.io/part-of: prow
+  namespace: test-pods
+  name: prowjob-private-sa
```

Comment:
comments of what permissions are allowed would probably help. Or at least the intended usage

Reply:
I've added the comments. Please take a second look.